applications/autonomous_worlds_on_substrate.md

```diff
@@ -0,0 +1,352 @@
+# SpellCrafter
+###### Autonomous Worlds and Conditional Access Control for Substrate Native Games
+
+- **Team Name:** Ideal Labs and Chainsafe Systems
+- **Funding Details:**
+  - **DOT**: For the **DOT** compensation, please provide a Polkadot address. (ex. multisig, with core contributors)
+
+## Game Project Overview :page_facing_up:
+
+# Game Project Application Form
+
+## 1. Pitch the Game
+
+### Please describe the game in a short pitch.
+
+> This is not only a pitch for a game, but for enabling verifiable onchain randomness, tlock, and verifiable computations for web3 games. The final output of this proposal will be a game, which Ideal Labs will develop in collaboration with Chainsafe Systems. 
+
+**Spellcrafter** is a "spell-crafting" card game where players discover ingredients, unlock recipes, and craft new spells in a challenge to be the most powerful mage. Players encounter offchain challenges (puzzles) that must be solved in order to unlock new recipes that can be used to craft spells on-chain. This game is a re-imagining of [Arcane Assembly](https://github.com/ArcaneAssemblers/spellcrafter), a game developed with Dojo (on Starknet), as a substrate native game. In this proposal, we also introduce new capabilities for games built on Polkadot: publicly verifiable on-chain randomness, timelock encryption, verifiable computation (zkSNARKs) and practical witness encryption. In the game, players must search the web, offchain, in order to learn knowledge that they can use to unlock recipes, such as by solving a puzzle or a riddle. 
+
+To enable spellcrafter on Substrate, we introduce publicly verifiable on-chain randomness, timelock encryption, and practical witness encryption capabilities to web3 gaming on Polkadot/Substrate. In collaboration with ChainSafe Gaming, Ideal Labs aims to enhance web3 gaming on Polkadot by integrating zkSNARKs via Circom and leveraging the Ideal Network's [post-finality gadget](https://medium.com/@ideal_labs/the-etf-post-finality-gadget-1dd6d7f12034), an MPC solution that enables a randomness beacon, to develop a practical witness encryption mechanism. This approach ensures fair play, enables asynchronous player coordination, and introduces proof-of-knowledge gating for in-game assets, events, or anything. Our framework uses off-chain computation with on-chain proof verification, using Groth 16, to provide new capabilities to Substrate-native games. This would not only allow for games like [zkHunt](https://0xparc.org/blog/zk-hunt) or [dark forest](https://zkga.me/) to be implemented on Substrate, but would allow for new on-chain conditional data access mechanisms to be created, such as requiring a zkSNARK of some in-game knowledge in order to proceed to the next level.

```

Comment:
> In this proposal, we also introduce new capabilities for games built on Polkadot: publicly verifiable on-chain randomness, timelock encryption, verifiable computation (zkSNARKs) and practical witness encryption. In the game, players must search the web, offchain, in order to learn knowledge that they can use to unlock recipes, such as by solving a puzzle or a riddle. 
>
> To enable spellcrafter on Substrate, we introduce publicly verifiable on-chain randomness, timelock encryption, and practical witness encryption capabilities to web3 gaming on Polkadot/Substrate.

It sounds repetitive. Not sure how to fix it by keeping the meaning but probably something like:

_To enable SpellCrafter on Substrate, we introduce these technologies—publicly verifiable on-chain randomness, timelock encryption, and practical witness encryption—into web3 gaming on Polkadot/Substrate._

Thought it does not avoid the repetitiveness, it acknowledges it :)

Reply:
I modified the text a bit, let me know what you think.